### DISCUSSION TITLE
Operator 2.2.0 Minimum Core Replicas 2

### DISCUSSION THREAD

**Discussion by erezblm** [emqx/emqx-operator#914] [Q&A]
Hey, I'm deploying the EMQX with the new operator 2.2.0 with the new apiversion "apps.emqx.io/v2beta1" 

and when deploying just a single core replicate i received the following error:
admission webhook "validator.apps.emqx.io" denied the request: the number of EMQX core nodes must be greater than 1

I understand the need from more than 1 replicas for production, but for the dev environment it helps us to save resources.

Is there any particular reason for the new restriction?

And another question, have you changed the infla ready state into "CoreNodesReady" instead of "Ready"? For some reason It's not changing into Ready, so my deployment is waiting for that although everything looks just fine (BTW i'm deploying 0 replicas for my development environment so it might be related.).

Thanks, Erez.

**Comment by Rory-Z** [comment]
Hi @erezblm The limitation on the number of Core nodes is only for high availability. If you do want to have only one Core node, a hack method is to directly modify the CRD file of EMQX to meet your requirements. However, I believe that in most cases, two Core nodes will not consume too many resources.

>  have you changed the infla ready state into "CoreNodesReady" instead of "Ready"?

No, you should just check "Ready", if didn't changing into Ready, maybe EMQX Operator have some bug. If you set volumeClaimTemplates in EMQX customer resource, then it's a know issue, I'm fixed it in 2.2.1-rc.1, if not, please open Operator's debug log and share it with me, you can append `--zap-devel=true` to EMQX Operator container's args, and check EMQX Operator log

**Comment by erezblm** [comment]
oh yes, i have used volumeClaimTemplates.
I suppose that for the 2.2.1-rc.1 I can't use the helm repository right?

Regarding the minimum 2 replicas, sure i've set it to 2 for now (even though they'll be running on the same node).
IMO it's better to leave the option to use a single replica for development environments

**Reply by Rory-Z** [reply]
You can install 2.2.1-rc.1 like this `helm install emqx-operator emqx/emqx-operator --namespace emqx-operator-system --create-namespace --version 2.2.1-rc.1`

**Reply by Rory-Z** [reply]
About minimum of core node, I will talk it with my colleague

**Reply by erezblm** [reply]
hey so i tried to deploy the 2.2.1-rc.1 with 2 cores and 0 replicants but now it's failed waiting on condition RepliantsProgressing.
Do we require to have replicants? if I just need 3 core replicas, i don't need any replicant, why do we require replicant node?
When I added a single replicant replica it worked just fine and state changed into Ready

**Reply by erezblm** [reply]
@Rory-Z , do you why is that?

Thanks, Erez.

**Reply by Rory-Z** [reply]
Sorry for delay, maybe this is a know issue, please check https://github.com/emqx/emqx-operator/issues/897#issuecomment-1674443103

**Comment by erezblm** [comment]
thanks you!

**Comment by Rory-Z** [comment]
Hi @erezblm the EMQX Operator 2.2.1 is released, in 2.2.1, EMQX Operator will no longer check the number of core nodes, and fixed https://github.com/emqx/emqx-operator/issues/897#issuecomment-1674443103, could you please try it :)

**Reply by erezblm** [reply]
amazing thank you very much i will try
